src/promptix/core/components/layer_composer.py

```
@@ -0,0 +1,550 @@
+
+        try:
+            template_obj = env.from_string(template)
+            result = template_obj.render(**variables)
+            # Convert escaped newlines to actual line breaks
+            return result.replace("\\n", "\n")
+        except TemplateError as e:
+            raise TemplateRenderError(
+                prompt_name=prompt_name,
+                template_error=str(e),
+                variables=variables
+            )
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)